### PR TITLE
fix: [CDS-43683]: Add null check before iterate the result list

### PR DESCRIPTION
**Pull request by Fernando-Dourado**
## Describe your changes
- Add null check before iterate the result list

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/37830)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/37830)

File: 930-delegate-tasks/src/main/java/software/wings/helpers/ext/artifactory/ArtifactoryServiceImpl.java [hunk 1]
```diff
@@ -83,7 +83,7 @@
 @BreakDependencyOn("io.harness.delegate.task.ListNotifyResponseData")
 public class ArtifactoryServiceImpl implements ArtifactoryService {
   private static final String REASON = "Reason:";
-  private static final String RESULTS = "results";
```
Comment (RajBaviskar) [inline 930-delegate-tasks/src/main/java/software/wings/helpers/ext/artifactory/ArtifactoryServiceImpl.java:86]: can we keep it private?

**Review by abhinav-harness — APPROVED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37830)* status: :shipit: complete! ✓

File: 930-delegate-tasks/src/main/java/software/wings/helpers/ext/artifactory/ArtifactoryServiceImpl.java (hunk 1 above)
Comment (Fernando-Dourado) [reply 930-delegate-tasks/src/main/java/software/wings/helpers/ext/artifactory/ArtifactoryServiceImpl.java:86]: The visibility was changed to package scope to use the same constant value inside the test class.

**Pull request merged**